src/v/archival/ntp_archiver_service.cc

```diff
@@ -314,13 +314,21 @@ ss::future<cloud_storage::download_result> ntp_archiver::download_manifest() {
     gate_guard guard{_gate};
     retry_chain_node fib(
       _manifest_upload_timeout, _cloud_storage_initial_backoff, &_rtcnode);
-    retry_chain_logger ctxlog(archival_log, fib, _ntp.path());
-    vlog(ctxlog.debug, "Downloading manifest for {}", _ntp);
     auto path = _manifest.get_manifest_path();
     auto key = cloud_storage::remote_manifest_path(
       std::filesystem::path(std::move(path)));
+    auto bucket = _bucket;
```

Comment:
Nit: I wonder if we should just initialize `_bucket` with the value from the config when running in the read replica mode. AFAIU read replica and archival modes are exclusive of each other so there won't be any problem with the archival loop using the wrong bucket.